benchmark/GasCostBenchmark.js

```diff
@@ -0,0 +1,298 @@
+const DocumentStore = artifacts.require("./DocumentStore.sol");
+const DocumentStoreWithRevokeReasons = artifacts.require("./DocumentStoreWithRevokeReasons.sol");
+const ProxyFactory = artifacts.require("./ProxyFactory.sol");
+const BaseAdminUpgradeabilityProxy = artifacts.require("./BaseAdminUpgradeabilityProxy.sol");
+
+DocumentStore.numberFormat = "String";
+const {generateHashes} = require("../scripts/generateHashes");
+
+const initializeAbi = {
+  constant: false,
+  inputs: [
+    {
+      internalType: "string",
+      name: "_name",
+      type: "string"
+    },
+    {
+      internalType: "address",
+      name: "owner",
+      type: "address"
+    }
+  ],
+  name: "initialize",
+  outputs: [],
+  payable: false,
+  stateMutability: "nonpayable",
+  type: "function"
+};
+
+const STORE_NAME = "THE_STORE_NAME";
+let lastIssuedHash = "0x3a267813bea8120f55a7b9ca814c34dd89f237502544d7c75dfd709a659f6432";
+const randomHashes = num => {
+  const generated = generateHashes(num, lastIssuedHash);
+  lastIssuedHash = generated[generated.length - 1];
+  return generated;
+};
+const randomHash = () => randomHashes(1)[0];
+
+describe("Gas Cost Benchmarks", () => {
+  const gasRecords = [];
+  const recordGasCost = (contract, context, gas) => {
+    gasRecords.push({
+      contract,
+      context,
+      gas
+    });
+  };
+  after(() => {
+    // eslint-disable-next-line no-console
+    console.table(gasRecords);
+  });
+
+  let staticDocumentStoreInstance;
+  let staticProxyFactoryInstance;
+  let staticDocumentStoreWithRevokeReasons;
+
+  before(async () => {
+    staticDocumentStoreInstance = await DocumentStore.new();
+    staticProxyFactoryInstance = await ProxyFactory.new();
+    staticDocumentStoreWithRevokeReasons = await DocumentStoreWithRevokeReasons.new();
+  });
+
+  contract("DocumentStore", accounts => {
+    let documentStoreInstance;
+    it("deploy", async () => {
```

Comment:
same as for normal test, u should put everything under the same it block

Reply:
Hmm it seemed like each transaction is independent of the other, don't think putting everything under one is good for readability.

Reply:
each test may be independant, but the result is on the overall (what if deploys is run after another it block, what if I skip deploy, etc.)